### PR TITLE
fix: only instantiate contract if contract address given

### PR DESCRIPTION
**Pull request by stephhuynh18**

### CODE REVIEW

**Review by v-stickykeys — COMMENTED** [review]
The error handling here seems a bit redundant and hard to follow.

Can we simplify to just

If usesmartcontractanchors { if contractaddress {...} else { throw } }

**Comment by stephhuynh18** [comment]
> The error handling here seems a bit redundant and hard to follow.
> 
> Can we simplify to just
> 
> If usesmartcontractanchors { if contractaddress {...} else { throw } }

@v-stickykeys yes it is =( I was being lazy HAHA because of the tests dislikes the option you provided. But i should do it right.

**Comment by gvelez17** [comment]
hey @stephhuynh18 should we fix conflicts and merge this during our on-call week?  figured cleanup old branches

File: src/services/blockchain/ethereum/ethereum-blockchain-service.ts [hunk 1]
```diff
@@ -151,10 +151,22 @@ export class EthereumBlockchainService implements BlockchainService {
...
+    if (
+      !this.config.blockchain.connectors.ethereum.contractAddress &&
+      this.config.useSmartContractAnchors
+    ) {
+      throw Error(
+        'You must set a smart contract address if you would like to use smart contract anchors'
+      )
+    }
+
+    if (this.config.blockchain.connectors.ethereum.contractAddress) {
```
Comment (gvelez17) [inline src/services/blockchain/ethereum/ethereum-blockchain-service.ts:164]: we don't need this if since we have the throw above?  but ya i like the clear error message
